src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -96,6 +96,9 @@ public int Execute()
 
             DisplayTestFiles();
 
+			// Throw a FileNotFoundException if any input file is missing.
+			CheckExistenceOfInputFiles(_options);
+
```

Comment:
I'm not sure how a filter would fit here. The runner doesn't know which files are runnable or even which exist. Although it's not in the current version, it may be asking the engine to kick off tests on some other machine. All it has is a string that may represent an assembly, may be a project containing assemblies or may be invalid. It has to wait to be told.